vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -196,6 +210,42 @@ type internal ProjectInfoManager
         | true, (_loadTime, originalOptions) -> Some originalOptions
         | _ -> this.TryGetOptionsForProject(projectId) 
 
+    [<Export>]
+    member this.HandleCommandLineChanges(path:string, sources:ImmutableArray<CommandLineSourceFile>, references:ImmutableArray<CommandLineReference>) =
+        let projectDir = Path.GetDirectoryName(path)
+        let fullPath p = if Path.IsPathRooted(p) then p else Path.Combine(projectDir, p)
+        let sourcePaths = sources |> Seq.map(fun s -> fullPath s.Path) |> Seq.toArray
+        let referencePaths = references |> Seq.map(fun r -> fullPath r.Reference) |> Seq.toArray
+
+        LogFile.log "======================================="
+        LogFile.log path 
+        LogFile.log projectDir
+        LogFile.log "======================================="
+        sourcePaths |> Seq.iter(fun s -> LogFile.log s)
+        LogFile.log "======================================="
+        referencePaths |> Seq.iter(fun r -> LogFile.log r)
+        LogFile.log "======================================="
+
+        lock lockObject (fun () -> match sourceFiles.TryGetValue path with
```

Comment:
Actually I believe you can just do ``sourceFiles.[path] <- sourcePaths`` - you don't need o explicitly Remove nor check if the collection already contains the value - the indexer will do what you need